test_app/tests/lib/utils/test_address.py

```
@@ -0,0 +1,32 @@
+import pytest
+
+from ansible_base.lib.utils.address import AddressType, AddressTypeResponse, classify_address
+
+
+@pytest.mark.parametrize(
+    "address,expected",
+    [
+        ("127.0.0.1", AddressTypeResponse(AddressType.IPv4, "127.0.0.1")),
+        ("[127.0.0.1]", AddressTypeResponse(AddressType.UNKNOWN)),
+        ("127.0.0.1:1234", AddressTypeResponse(AddressType.IPv4, "127.0.0.1", "1234")),
+        ("10.0.1.1:1234", AddressTypeResponse(AddressType.IPv4, "10.0.1.1", "1234")),
+        ("10.0.1.1", AddressTypeResponse(AddressType.IPv4, "10.0.1.1")),
+        ("[10.0.1.1]", AddressTypeResponse(AddressType.UNKNOWN)),
+        ("::1", AddressTypeResponse(AddressType.IPv6, "::1")),
+        ("[::1]", AddressTypeResponse(AddressType.HOSTNAME, "[::1]")),
+        ("[::1]:1234", AddressTypeResponse(AddressType.HOSTNAME, "[::1]", "1234")),
+        ("2600:1f18:218b:5902:e5d4:54de:fdc1:24b8", AddressTypeResponse(AddressType.IPv6, "2600:1f18:218b:5902:e5d4:54de:fdc1:24b8")),
+        ("[2600:1f18:218b:5902:e5d4:54de:fdc1:24b8]", AddressTypeResponse(AddressType.HOSTNAME, "[2600:1f18:218b:5902:e5d4:54de:fdc1:24b8]")),
+        ("2600:1f18:218b:5902:e5d4:54de:fdc1:24b8:1234", AddressTypeResponse(AddressType.IPv6, "2600:1f18:218b:5902:e5d4:54de:fdc1:24b8", "1234")),
```

This should throw an error imho. No unbracketed IPv6+Port combination should be fed to Gateway in the first place. And if it is, throw an error instead of trying to accommodate for that.

---

As discussed with @PabloHiro in the IPv6 WG channel unless/until the current mixture of IPv6 specification is standardized and implemented we cannot impose a standard here without potentially breaking deployments.